azure/datalake/store/transfer.py

```diff
@@ -111,6 +111,35 @@ def __str__(self):
 Chunk = namedtuple('Chunk', 'name state offset expected actual exception')
 
 
+class NoSerLock(object):
+    """
+    Pickle friendly wrapper for threading.Lock (which is non-inheritable
+     so we wrap relevant functions)
+    The main part of this wrapper is that __getstate__ will return an empty
+    tuple so that the actual lock isn't serialized
```

Comment:
If this class is only used in a single class then would it make more sense to define `__getstate__()` for that class instead?